File: Java/2444.java
class Solution {
    public long countSubarrays(int[] nums, int minK, int maxK) {
        long result = 0;
        int subArrayStartIndex = 0;
        int latestMinIndex = -1;
        int latestMaxIndex = -1;

        for (int i = 0; i < nums.length; i++) {
            // invalid sub array, start over
            if (nums[i] < minK || nums[i] > maxK) {
                latestMinIndex = -1;
                latestMaxIndex = -1;
                subArrayStartIndex = i + 1;
            }
            if (nums[i] == minK) {
                latestMinIndex = i;
            }
            if (nums[i] == maxK) {
                latestMaxIndex = i;
            }
            result += Math.max(0L, Math.min(latestMinIndex, latestMaxIndex) - subArrayStartIndex + 1);
        }
        return result;
    }
}

File: Python/2444.py
from typing import List


class Solution:
    def countSubarrays(self, nums: List[int], minK: int, maxK: int) -> int:
        result = 0
        subarr_start_index = prev_min_index = prev_max_index = -1

        for i, num in enumerate(nums):
            if num < minK or num > maxK:
                subarr_start_index = i
            if num == minK:
                prev_min_index = i
            if num == maxK:
                prev_max_index = i

            result += max(0, min(prev_min_index, prev_max_index) - subarr_start_index)

        return result
